internal/otlpjson/databuffer.go

```diff
@@ -0,0 +1,239 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpjson // import "go.opentelemetry.io/collector/internal/otlpjson"
+
+import (
+	"bytes"
+	"encoding/base64"
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+type dataBuffer struct {
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/model/otlp/json_marshaler.go

Reply:
Already implemented, don't need to re-implement this.

Reply:
Thanks, didn’t notice that a JSON serializer already exists. Unfortunately, the format is not “greppable” since a single line contains many spans (and that’s also the issue with `otlptext`, except that spans also take multiple lines). Maybe we can rename `otlpjson` to `loggingexporter/jsonformat` to indicate that it is a format used specifically for logging?